roles/provision-unifiedpush-apb/tasks/provision-ups.yml

```diff
@@ -1,3 +1,60 @@
+- name: create UPS service
+  k8s_v1_service:
+    name: '{{ ups_service_name }}'
+    namespace: '{{ namespace }}'
+    annotations:
+      org.aerogear.metrics/plain_endpoint: "/rest/prometheus/metrics"
+    labels:
+      app: ups
+      service: ups
+      mobile: enabled
+    selector:
+      app: ups
+      service: ups
+    ports:
+      - name: web
+        port: 80
+        target_port: 8080
+  register: unifiedpush_service
+
+- name: create UPS Proxy
+  k8s_v1_service:
+    name: '{{ ups_proxy_service_name }}'
+    namespace: '{{ namespace }}'
+    labels:
+      app: ups
+      service: ups-proxy
+    selector:
+      app: ups
+      service: ups
+    ports:
+      - name: web
+        port: 80
+        target_port: '{{ ups_proxy_port }}'
+
+- name: create ups https route
+  openshift_v1_route:
+    name: '{{ ups_route_name }}'
+    namespace: '{{ namespace }}'
+    labels:
+      app: ups
+      service: ups-proxy
+    to_name: ups-proxy
+    spec_port_target_port: web
+    spec_tls_termination: edge
+  register: ups_route
+
+- name: "Create ups secret yaml file"
+  template:
+    src: secret.yml.j2
+    dest: /tmp/secret.yaml
+
+- name: "Create UPS secret"
```

Comment:
This secret is used by the ups-config-operator, so my thinking is that it should get provisioned _before_ the ups pod. Unfortunately, this doesn't work here, because the namespace PushApplication ID is one of the things that it's supposed to be storing (which needs UPS (incl. the config operator) to be running _first_.

I have seen this error a couple of times today in an openshift dedicated environment after needing to restart UPS because postgresql wasn't ready so it crashed and I needed to redeploy:

```
oc logs ups-1-5fq94 -c ups-sync-service 2019/06/18 15:42:34 Entering watch loop 2019/06/18 15:42:44 Error creating push client: secrets "unified-push-server" not found panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xe454c8] goroutine 6 [running]: github.com/aerogear/ups-config-operator/pkg/configOperator.(*UpsClientImpl).getApplicationId(0xc420330b40, 0x116ed60, 0xc420330b40) /go/src/github.com/aerogear/ups-config-operator/pkg/configOperator/upsClient.go:232 +0x8 github.com/aerogear/ups-config-operator/pkg/configOperator.ConfigOperator.compareUPSVariantsWithClientConfigs(0x114ce60, 0xc4203522c0, 0x114ff00, 0xc42000e6e0, 0x116c240, 0xc4203522d0) /go/src/github.com/aerogear/ups-config-operator/pkg/configOperator/configOperator.go:123 +0x79 github.com/aerogear/ups-config-operator/pkg/configOperator.ConfigOperator.startPollingUPS(0x114ce60, 0xc4203522c0, 0x114ff00, 0xc42000e6e0, 0x116c240, 0xc4203522d0) /go/src/github.com/aerogear/ups-config-operator/pkg/configOperator/configOperator.go:58 +0x67 created by github.com/aerogear/ups-config-operator/pkg/configOperator.ConfigOperator.StartService /go/src/github.com/aerogear/ups-config-operator/pkg/configOperator/configOperator.go:43 +0x96
```

@secondsun @danielpassos @pb82 @david-martin if any of you have any idea what's going on there, and any idea how this is supposed to work, please let us know! :)

Reply:
I have removed this commit now, in the hopes that the issue I was trying to solve with it is really caused by the issue solved by the remaining commits. It is a confusing thing to have to troubleshoot though, so I welcome any feedback around how it's supposed to work.

Reply:
I wasn't working on the original apb stuff so I don't know, but I think maybe it works by just crashing and restarting until everything else is ready?